code/modules/mod/modules/module_holding.dm

```diff
@@ -0,0 +1,99 @@
+#define HOLDING_MODULE_PREVENT_DUPLICATE_CHECK 1
+#define HOLDING_MODULE_CHECK_CONFIRMED 2
+
+/obj/item/mod/module/storage/holding
+	name = "MOD storage module of holding"
+	desc = "A prototype storage module utilizing the power of anomalous bluespace phenomena \
+		to store copious amounts of matter. Unfortunately, it suffers from the same drawbacks as its standalone counterpart, \
+		including <b>tearing catastrophic rifts in reality</b> when nested inside bluespace pockets produced through similar means."
+	icon_state = "antigrav"
+	complexity = 4
+	storage_type = null // core-less modules should be safe to insert into bags of holding
+	var/prebuilt = FALSE
+	var/core_removable = TRUE
+	var/datum/component/anomaly_locked_module/anomalock
+	var/list/possible_bag_bombs = list()
+
+/obj/item/mod/module/storage/holding/Initialize(mapload)
+	. = ..()
+	anomalock = AddComponent(/datum/component/anomaly_locked_module,\
+		list(/obj/item/assembly/signaler/anomaly/bluespace),\
+		prebuilt,\
+		core_removable,\
+		PROC_REF(pre_core_inserted),\
+		PROC_REF(on_core_inserted),\
+		PROC_REF(on_core_removed),\
+	)
+	RegisterSignal(src, COMSIG_MODULE_TRY_INSTALL, PROC_REF(try_install))
+
+/obj/item/mod/module/storage/holding/Destroy()
+	. = ..()
+	possible_bag_bombs.Cut()
+
+/obj/item/mod/module/storage/holding/proc/pre_core_inserted(mob/user, obj/item/core, list/modifiers)
+	if(possible_bag_bombs[core] == HOLDING_MODULE_PREVENT_DUPLICATE_CHECK)
+		return ITEM_INTERACT_FAILURE
+	var/datum/storage/bag_of_holding/other_bag
+	for(var/atom/nested_loc in get_nested_locs(src))
+		if(istype(nested_loc.atom_storage, /datum/storage/bag_of_holding))
+			other_bag = nested_loc.atom_storage
+			break
+	if(other_bag)
+		if(possible_bag_bombs[core] == HOLDING_MODULE_CHECK_CONFIRMED)
+			other_bag.create_rift(core, user)
+		else
+			INVOKE_ASYNC(src, PROC_REF(recursive_core_insertion), other_bag, core, user, modifiers)
+		return ITEM_INTERACT_BLOCKING
+
+/obj/item/mod/module/storage/holding/proc/recursive_core_insertion(datum/storage/bag_of_holding/bag_storage, obj/item/core, mob/user, list/modifiers)
+	possible_bag_bombs[core] = HOLDING_MODULE_PREVENT_DUPLICATE_CHECK
+	if(bag_storage.confirm_recursive_insertion(core, user) && !QDELETED(core) && user.is_holding(core) && IsReachableBy(user))
+		possible_bag_bombs[core] = HOLDING_MODULE_CHECK_CONFIRMED
+		base_item_interaction(user, core, modifiers)
```

Comment:
```suggestion
		bag_storage.attempt_insert(core, user)
```
I assume you are trying to insert the core into the storage. If so please don't call the entire attack chain for it